spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/NamedConfigMapContextToSourceDataProvider.java

```diff
@@ -42,8 +43,21 @@ public KubernetesClientContextToSourceData get() {
 			NamedConfigMapNormalizedSource source = (NamedConfigMapNormalizedSource) context.normalizedSource();
 
 			return new NamedSourceData() {
+				@Override
+				protected String generateSourceName(String target, String sourceName, String namespace,
+						String[] activeProfiles) {
+					if (source.appendProfileToName()) {
+						return ConfigUtils.sourceName(target, sourceName, namespace, activeProfiles);
+					}
+					return super.generateSourceName(target, sourceName, namespace, activeProfiles);
+				}
+
 				@Override
 				public MultipleSourcesContainer dataSupplier(LinkedHashSet<String> sourceNames) {
+					if (source.appendProfileToName()) {
```

Comment:
Yes both of those assumptions are correct.

Reply:
I agree that the flag `appendProfileToName` is a bit overloaded in the PR.  It is actually the final thing I was trying to make better in the PR.

I thought about an approach similar to what you are proposing here.  Ultimately I decided against it because who is to say that next "Caller" does not want to also do the same thing the config server needs?  In which case having a "caller" API seems to be a bit fragile.

What I dislike about my approach now is that you could also imaging a situation where you want to append the profile to the property source name, but don't want to exclude the default profile.  Right now both pieces of functionality are tied to the one flag `appendProfileToName`.

It actually feels more "right" to me if we modify `KubernetesClientConfigContext` and add more to the "context to compute property sources".  I think here it might make sense to add a flag to include the default profile.